src/structstore_pybind.cpp

```diff
@@ -50,6 +50,18 @@ py::object to_object(const StructStoreField& field) {
 }
 
 static void from_object(FieldAccess access, const py::handle& value, const std::string& field_name) {
+    if (py::isinstance<List>(value)
+            && access.get_type() == FieldTypeValue::LIST) {
+        if (value.cast<List&>() == access.get<List>()) {
```

Comment:
I'm not comfortable with calling the comparison operator here, which might recurse through the whole list structure if we change the comparison operator implementation at some point. I guess explicitly comparing pointers here (and for StructStore instances below) shows more obviously what is intended here.

Reply:
Ok, using only pointer comparisons.